Source/Models/APIClient.swift

```diff
@@ -2,19 +2,24 @@ import Foundation
 import Argo
 import Result
 
-public struct APIClient {
+public struct APIClient: Client {
   private let requestPerformer: RequestPerformer
 
+  public var queue: dispatch_queue_t? = dispatch_get_main_queue()
```

Comment:
I don't think this should be mutable. That would introduce the possibility of performing a request and changing the queue before it returns, resulting in unexpected behavior. Can we instead pass this in via `init`?